packages/concerto-core/test/introspect/stringvalidator.js

```diff
@@ -36,22 +48,51 @@ describe('StringValidator', () => {
 
         it('should throw for invalid regexes', () => {
             (() => {
-                new StringValidator(mockField, { pattern: '^[A-z' });
+                new StringValidator(mockField, { pattern: '^[A-z' }, VALID_MIN_LENGTH_AND_MAX_LENGTH_AST);
             }).should.throw(/Validator error for field/);
         });
 
+        it('should throw for invalid string length', () => {
+            (() => {
+                new StringValidator(mockField, { pattern: '^[A-z]' }, NO_PARMS_IN_AST);
+            }).should.throw(/Invalid string length, minLength and-or maxLength must be specified/);
+        });
+
+        it('should throw for string min length is bigger than max length', () => {
+            (() => {
+                new StringValidator(mockField, { pattern: '^[A-z]' }, MIN_LENGTH_IS_HIGHER_THAN_MAX_LENGTH);
+            }).should.throw(/minLength must be less than or equal to maxLength/);
+        });
+
+        it('should throw for negetive string min length', () => {
+            (() => {
+                new StringValidator(mockField, null, NEGETIVE_MIN_LENGTH);
+            }).should.throw(/minLength and-or maxLength must be positive integers/);
+        });
+
+        it('should throw for negetive string max length', () => {
+            (() => {
+                new StringValidator(mockField, null, NEGETIVE_MAX_LENGTH);
+            }).should.throw(/minLength and-or maxLength must be positive integers/);
+        });
+
+        it('should throw for negetive string length', () => {
+            (() => {
+                new StringValidator(mockField, null, NEGETIVE_LENGTH);
+            }).should.throw(/minLength and-or maxLength must be positive integers/);
+        });
     });
 
     describe('#validate', () => {
 
         it('should ignore a null string', () => {
-            let v = new StringValidator(mockField, { pattern: '^[A-z][A-z][0-9]{7}' });
+            let v = new StringValidator(mockField, { pattern: '^[A-z][A-z][0-9]{7}' }, VALID_MIN_LENGTH_AND_MAX_LENGTH_AST);
```

Comment:
Why do we need to change the existing tests? It's important that if we skip this parameter, then we should get the same behaviour.